src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureFilter.cpp

```diff
@@ -360,12 +645,247 @@ namespace OpenMS
     }
   }
 
-  std::map<String,int> MRMFeatureFilter::countLabelsAndTransitionTypes(
+  void MRMFeatureFilter::EstimateDefaultMRMFeatureQCValues(const std::vector<FeatureMap>& samples, MRMFeatureQC& filter_template, const TargetedExperiment& transitions, const bool& init_template_values)
+  {
+    // iterature through each sample and accumulate the min/max values in the samples in the filter_template
+    for (size_t sample_it = 0; sample_it < samples.size(); sample_it++) {
+
+      // iterate through each component_group/feature
+      for (size_t feature_it = 0; feature_it < samples.at(sample_it).size(); ++feature_it)
+      {
+        String component_group_name = (String)samples.at(sample_it).at(feature_it).getMetaValue("PeptideRef");
+        std::map<String, int> labels_and_transition_types = countLabelsAndTransitionTypes(samples.at(sample_it).at(feature_it), transitions);
+
+        // iterate through each component/sub-feature
+        for (size_t sub_it = 0; sub_it < samples.at(sample_it).at(feature_it).getSubordinates().size(); ++sub_it)
+        {
+          String component_name = (String)samples.at(sample_it).at(feature_it).getSubordinates().at(sub_it).getMetaValue("native_id");
+
+          // iterate through multi-feature/multi-sub-feature QCs/filters
+          // iterate through component_groups
+          for (size_t cg_qc_it = 0; cg_qc_it < filter_template.component_group_qcs.size(); ++cg_qc_it)
+          {
+            if (filter_template.component_group_qcs.at(cg_qc_it).component_group_name == component_group_name)
+            {
+              const double rt = samples.at(sample_it).at(feature_it).getRT();
+              if (sample_it == 0 && init_template_values) {
+                initRange(rt,
+                  filter_template.component_group_qcs.at(cg_qc_it).retention_time_l,
```

Comment:
all occurences of `filter_template.component_group_qcs.at(cg_qc_it)` could be replace by `somevar` if the loop would be range-based. Much easier to understand and probably faster (if thats a concern).
`for (auto& somevar: filter_template.component_group_qcs) ...`

Reply:
The iterator was used because at certain points we were working with two aligned lists.  We have tested the speed using the range based for loop and it did not make a difference at least for the particular logic we were using and the data sets we were running.